File: Android/app/src/main/java/kitchensv/com/models/Order.kt
package kitchensv.com.models

import com.google.firebase.Timestamp
import com.google.firebase.firestore.ServerTimestamp

data class Order(
    var id: String = "",
    var numeroMesa: String = "",
    var estado: String = "EN PROCESO", // EN PROCESO, ENTREGADA, PAGADA
    var platos: List<Plato> = emptyList(),
    var notas: String = "",
    @ServerTimestamp var horaCreacion: Timestamp? = null
)

data class Plato(
    var referenciaProducto: String = "", // ID del producto en Firebase
    var nombreProducto: String = "", // Nombre para mostrar
    var cantidad: Int = 1
)

data class Producto(
    var id: String = "",
    var nombre: String = "",
    var precio: Double = 0.0,
    var categoria: String = "",
    var disponible: Boolean = true
) {
    override fun toString(): String {
        return nombre
    }
}

File: React/src/types/Order.ts
import { Timestamp } from "firebase/firestore";

export type OrderItem = {
  nombreProducto: string;
  cantidad: number;
  referenciaProducto?: string;
};

export type OrderStatus =
  | "PENDIENTE"
  | "PREPARANDO"
  | "PAUSADA"
  | "LISTO"
  | "ENTREGADA";

export type Order = {
  id?: string; // ID de Firestore
  numeroMesa: string;
  notas?: string;
  platos: OrderItem[];
  estado: OrderStatus;
  horaCreacion: Timestamp; // Timestamp de Firestore
  horaCocina?: Timestamp; // Timestamp de Firestore
  type: "dine_in" | "take_off";
};
